### DISCUSSION TITLE
Disable router feature on some devices

### DISCUSSION THREAD

**Discussion by grobihh** [Koenkk/zigbee2mqtt#20208] [Devices]
I would expect that this has been discussed already but I could not find any topics.

I have a Sonoff S26R2ZB (Switch, firmware 2.1.1) which also acts as a ZigBee router.
Obviously that model has some issues. With firmware 2.1.1 the connection seems to be stable.
https://github.com/Koenkk/zigbee2mqtt/issues/10282

My problem is that it acts badly as a router.
When S26R2ZB is plugged in some devices will chose that switch as their router. Then those devices will be hardly available in my ZigBee network.

**Is there a way of disabling the Router function per device?**
I would like to use S26R2ZB only as an end device. In that role it does work fine currently.

**Comment by grobihh** [comment]
In case anybody cares, situation seems to be better when disabling "Optimistic" on the Sonoff S26R2ZB switch.

**Comment by yozik04** [comment]
I would appreciate this feature as well. I possess a physical switch that turns off all the lights in my house. Connected to this circuit is a Zigbee switch, which consistently attempts to function as a router. When I turn off the lights, the switch becomes unavailable, forcing other devices to find new routes. This process is time-consuming, and the devices remain unavailable for a duration. I want to prevent this by disallowing other devices from using this switch as a router.

**Comment by Koenkk** [comment]
This is not possible, the coordinator has no control over wether a device acts as an end device or router.

**Reply by yozik04** [reply]
It's unfortunate. However, I appreciate your response.

**Reply by grobihh** [reply]
Seems that I solved the problem by installing a dedicated repeater in the room where the problematic switch sits.
https://www.zigbee2mqtt.io/devices/TS0207_repeater.html#tuya-ts0207_repeater
